tracee-rules/benchmark/signature/rego/aio.rego

```diff
@@ -0,0 +1,79 @@
+package tracee.aio
+
+import data.tracee.helpers
+
+__rego_metadoc__ := {
+    "id": "TRC-AIO",
+    "version": "0.1.0",
+    "name": "All in one Tracee signature",
+    "description": "Combines Anti-Debugging and Code-injection",
+    "tags": ["linux", "container"],
+    "properties": {
+        "Severity": 3,
+        "MITRE ATT&CK": "Defense Evasion: Execution Guardrails",
+    }
+}
+
+# anti_debugging
+tracee_anti_debugging_tracee_selected_events[eventSelector] {
+	eventSelector := {
+		"source": "tracee",
+		"name": "ptrace"
+	}
+}
```

Comment:
In this PR this file is hand generated to include prefixes like `tracee_anti_debugging_` prior to each rego stanza to keep them separate. Now if this was an approach we ultimately end up taking we could do something of the following (thanks to @danielpacak for brainstorming earlier):

1. Keep the same contract we have today to accept Rego signatures from users.
2. During the first run, parse all rego files (signatures) we have and combine them into one file. We could potentially save this in the /tmp/tracee directory for re-use
3. Each signatures rego policy is kept unique within the combined rego file by prefixing it with the prefixes as seen here so that we can determine which rule was triggered within this single AIO policy.
4. We would probably need some rework to figure out which rule within the combined Rego file was triggered during evaluation in the tracee-rules processing side.

Reply:
What do you think about this @itaysk – to me steps 1, 2 and 3 seem doable but step 4 is a little unclear as I'm not an expert on Rego policies. Is it something we can do with Rego?

Reply:
Thanks for sharing this is very interesting. @danielpacak discussed this with me earlier this week and I told him that I don't think we need to merge the rules, we can just load them into the same rego instance which will presumably give the same effect. no rewrite is needed since the rules are namespaced already.

Reply:
I also told there are possible downsides to unifying rego instances:
1. reduced parallelism, the entire rego engine becomes a big bottleneck.
2. the entire "dispatching" mechanism which is supposed to deliver the right events to the relevant signatures becomes useless if there's one big signature that is receiving all the events. (I believe this benchmark doesn't account for that). 

An improvement suggestion: don't unify all signatures into one rego instance, but unify all signatures with the same event selectors (one per signaturesIndexMap item).

Reply:
> I also told there are possible downsides to unifying rego instances:
> 
> 1. reduced parallelism, the entire rego engine becomes a big bottleneck.
> 2. the entire "dispatching" mechanism which is supposed to deliver the right events to the relevant signatures becomes useless if there's one big signature that is receiving all the events. (I believe this benchmark doesn't account for that).
> 

It does as it can be seen here: https://github.com/aquasecurity/tracee/pull/825/#issue-690256492 - the graph is an overall CPU utilization. 

> An improvement suggestion: don't unify all signatures into one rego instance, but unify all signatures with the same event selectors (one per signaturesIndexMap item).

As discussed earlier offline this real world test effectively shows a performance improvement https://github.com/aquasecurity/tracee/pull/825/#issuecomment-881782164